### PR TITLE
feat: Implement if_to_bool_then assist

### PR DESCRIPTION
**Pull request by Veykril**
One half of https://github.com/rust-analyzer/rust-analyzer/issues/8413

### CODE REVIEW

**Comment by Veykril** [comment]
@lnicola do you have mixed feelings about this assist or why the reaction?

**Comment by lnicola** [comment]
Don't mind me, I guess I'm not a fan of combinators like `bool::then`. The code looks good, I think.

**Comment by lnicola** [comment]
bool_then

**Comment by matklad** [comment]
> I guess I'm not a fan of combinators like bool::then

Me too! I generally prefer explicit matches and ifs over combinators. But I guess if t-libs thinks that `.then` is a good idea, we should support it in the tooling.

File: crates/syntax/src/ast/node_ext.rs [hunk 1]
```diff
@@ -56,6 +56,10 @@ impl ast::BlockExpr {
     pub fn is_empty(&self) -> bool {
         self.statements().next().is_none() && self.tail_expr().is_none()
     }
+
+    pub fn as_lone_tail(&self) -> Option<ast::Expr> {
```
Comment (matklad) [inline crates/syntax/src/ast/node_ext.rs:60]: I don't think that's the best place for this functionality -- one of the constraints of AST is that it should provide only "general" functionality. Specific helpers should generally be defined in the IDE layer. 

See, eg, `unwrap_trivial_block`. 

I suggest moving this elsewhere (and maybe amending the doc comment at the top explaining what should, and what should not go here).

**Pull request merged**